src/log-store/src/kafka/client_manager.rs

```diff
@@ -80,7 +84,11 @@ impl ClientManager {
             base: config.backoff.base as f64,
             deadline: config.backoff.deadline,
         };
-        let client = ClientBuilder::new(config.broker_endpoints.clone())
+        let mut broker_endpoints = Vec::with_capacity(config.broker_endpoints.len());
+        for endpoint in &config.broker_endpoints {
+            broker_endpoints.push(Self::resolve_broker_endpoint(endpoint).await?);
+        }
```

Comment:
Functional programming style is more preferable.

Reply:
But I don't know how to handle error early with fp style, i.e. stop resolve and return error instantly,if an error occurs.